impeller/entity/contents/filters/directional_gaussian_blur_filter_contents_unittests.cc

```diff
@@ -0,0 +1,185 @@
+  EXPECT_CALL(*texture, GetSize()).WillRepeatedly(Return(ISize(100, 100)));
+  FilterInput::Vector inputs = {FilterInput::Make(texture)};
+  Entity entity;
+  entity.SetTransformation(Matrix::MakeTranslation({100, 100, 0}));
+  std::optional<Rect> coverage = contents->GetFilterCoverage(
+      inputs, entity, /*effect_transform=*/Matrix::MakeScale({2.0, 2.0, 1.0}));
+  ASSERT_EQ(coverage, Rect::MakeLTRB(100 - 2, 100, 200 + 2, 200));
+}
+
+TEST_F(DirectionalGaussianBlurFilterContentsTest, FilterSourceCoverage) {
+  Scalar sigma_radius_1 = CalculateSigmaForBlurRadius(1.0);
+  auto contents = std::make_unique<DirectionalGaussianBlurFilterContents>();
+  contents->SetSigma(Sigma{sigma_radius_1});
+  contents->SetDirection({1.0, 0.0});
+  std::optional<Rect> coverage = contents->GetFilterSourceCoverage(
+      /*effect_transform=*/Matrix::MakeScale({2.0, 2.0, 1.0}),
+      /*output_limit=*/Rect::MakeLTRB(100, 100, 200, 200));
+  ASSERT_EQ(coverage, Rect::MakeLTRB(100 - 2, 100, 200 + 2, 200));
+}
+
+TEST_F(DirectionalGaussianBlurFilterContentsTest, RenderNoCoverage) {
+  Scalar sigma_radius_1 = CalculateSigmaForBlurRadius(1.0);
+  auto contents = std::make_unique<DirectionalGaussianBlurFilterContents>();
+  contents->SetSigma(Sigma{sigma_radius_1});
+  contents->SetDirection({1.0, 0.0});
+  auto mock_context = std::make_shared<MockImpellerContext>();
+  auto mock_typographer_context = std::make_shared<MockTypographerContext>();
+  auto mock_allocator = std::make_shared<MockAllocator>();
+  auto mock_render_target_allocator =
+      std::make_shared<MockRenderTargetAllocator>(mock_allocator);
+  ContentContext renderer(mock_context, mock_typographer_context,
+                          mock_render_target_allocator);
+  Entity entity;
+  Rect coverage_hint = Rect::MakeLTRB(0, 0, 0, 0);
+  std::optional<Entity> result =
+      contents->GetEntity(renderer, entity, coverage_hint);
+  ASSERT_FALSE(result.has_value());
+}
+
+TEST_F(DirectionalGaussianBlurFilterContentsTest, RenderSomething) {
+  SetupMinimalContext();
+  TextureDescriptor desc = {
+      .size = ISize(100, 100),
+  };
+  std::shared_ptr<MockTexture> texture = std::make_shared<MockTexture>(desc);
+  EXPECT_CALL(*texture, GetSize()).WillRepeatedly(Return(ISize(100, 100)));
+  Scalar sigma_radius_1 = CalculateSigmaForBlurRadius(1.0);
+  auto contents = std::make_unique<DirectionalGaussianBlurFilterContents>();
+  contents->SetSigma(Sigma{sigma_radius_1});
+  contents->SetDirection({1.0, 0.0});
+  contents->SetInputs({FilterInput::Make(texture)});
+  ContentContext renderer(mock_context_, mock_typographer_context_,
+                          mock_render_target_allocator_);
+  Entity entity;
+  Rect coverage_hint = Rect::MakeLTRB(0, 0, 0, 0);
+  std::optional<Entity> result =
+      contents->GetEntity(renderer, entity, coverage_hint);
+  ASSERT_TRUE(result.has_value());
+  ASSERT_EQ(result.value().GetBlendMode(), BlendMode::kSourceOver);
```

Comment:
I would like to have better asserts here about the coverage of the result.  I'm not 100% up to date about what the values should be considering the coverage hint yet though.